### PR TITLE
Port loolwsd-systemplate-setup to FreeBSD.

### PR DESCRIPTION
**Pull request by arrowd**
Signed-off-by: Gleb Popov <6yearold@gmail.com>
Change-Id: I9944e79d0fe220a5f2fe40d2ba98619c22d25a77

* Target version: master 

- [X] Code is properly formatted
- [X] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [X] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: loolwsd-systemplate-setup [hunk 1]
```diff
@@ -21,6 +21,16 @@ CHROOT=`cd $CHROOT && /bin/pwd`
 INSTDIR_LOGICAL=`cd $INSTDIR && /bin/pwd -L`
 INSTDIR=`cd $INSTDIR && /bin/pwd -P`
 
+if [ ! `uname -s` = "FreeBSD" ]; then
+    CP=cp
+    REALPATH=realpath
+    LOCALBASE=usr/
+else
+    CP=gcp
```
Comment (Ashod) [inline loolwsd-systemplate-setup:29]: Freebsd doesn't have `cp`? (I find it hard to believe.)

File: loolwsd-systemplate-setup [hunk 2]
```diff
@@ -55,7 +65,11 @@ find etc/fonts \
 # they link to.
 find $INSTDIR -name 'xpdfimport' |
 while read file; do
-    ldd $file 2>/dev/null
+    if [ ! `uname -s` = "FreeBSD" ]; then
+        ldd $file 2>/dev/null
+    else
+        ldd $file 2>/dev/null | tail -n +2
```
Comment (Ashod) [inline loolwsd-systemplate-setup:71]: A comment to explain what's going on?

**Review by Ashod — COMMENTED** [review]
Thanks a lot Gleb! A couple of questions inline. (Also, we need to get Jenkins happy. A rebase perhaps will help.)

File: loolwsd-systemplate-setup (hunk 1 above)
Comment (arrowd) [reply loolwsd-systemplate-setup:29]: It does, but it has different flags. Some like `--dereference` are missing. So I thought it is simpler to GNU `cp`.

**Comment by Ashod** [comment]
@timar would be great if you can take a look and see if you have concerns. I'd personally prefer to avoid (or minimize) the conditionals on system name as much as possible.

Thanks.

File: loolwsd-systemplate-setup [hunk 3]
```diff
@@ -55,7 +65,16 @@ find etc/fonts \
 # they link to.
 find $INSTDIR -name 'xpdfimport' |
 while read file; do
-    ldd $file 2>/dev/null
+    if [ ! `uname -s` = "FreeBSD" ]; then
+        ldd $file 2>/dev/null
+    else
+        # lld on FreeBSD prints extraneous line, so chop it
```
Comment (timar) [inline loolwsd-systemplate-setup:71]: lld > ldd

Comment (timar) [reply loolwsd-systemplate-setup:71]: but why is this necessary? The grep in line 79 won't match the empty line.

**Review by timar — CHANGES_REQUESTED** [review]

File: loolwsd-systemplate-setup (hunk 3 above)
Comment (arrowd) [reply loolwsd-systemplate-setup:71]: It is not empty but

    /usr/local/lib/libreoffice/program/xpdfimport:

**Comment by timar** [comment]
> @timar would be great if you can take a look and see if you have concerns. I'd personally prefer to avoid (or minimize) the conditionals on system name as much as possible.
> 
> Thanks.

@ash I tend to agree. We could use autotools to generate this script. But that'd be an overkill. :)

File: loolwsd-systemplate-setup (hunk 3 above)
Comment (timar) [reply loolwsd-systemplate-setup]: > It is not empty but
> 
> ```
> /usr/local/lib/libreoffice/program/xpdfimport:
> ```

Oh... Would  `cut -s -d " " -f 3` help then?

**Comment by mmeeks** [comment]
Oooh - I suspect we don't use xpdfimport anymore - with the built-in pdfium code. Probably dunging that out would be useful (with some testing around PDF of course).

File: loolwsd-systemplate-setup (hunk 3 above)
Comment (arrowd) [reply loolwsd-systemplate-setup]: Yep, seems to be working!

**Review by timar — APPROVED** [review]
Thanks!

**Comment by timar** [comment]
> Oooh - I suspect we don't use xpdfimport anymore - with the built-in pdfium code. Probably dunging that out would be useful (with some testing around PDF of course).

@mmeeks Collabora's releases do not include xpdfimport (wrapper for the GPL-ed libpoppler). But technically it's possible to configure it.

**Pull request merged**